### PR TITLE
adding new school changes

### PR DESCRIPTION
**Pull request by SoLetsDev**

### CODE REVIEW

File: backend/src/components/institute/institute.js [hunk 1]
```diff
@@ -241,6 +241,103 @@ async function getCachedSchoolBySchoolID(req, res) {
...
+async function addSchool(req, res) {
+  try {
+    const token = getBackendToken(req);
+    let hasSchoolAdminRoleAuth;
+
+    if(req.body.schoolCategoryCode === 'INDEPEND' || req.body.schoolCategoryCode === 'INDP_FNS'){
+      hasSchoolAdminRoleAuth = req.session.roles.includes('SCHOOL_ADMIN') || req.session.roles.includes('SCHOOL_INDEPENDENT_ADMIN');
+    } else {
+      hasSchoolAdminRoleAuth = req.session.roles.includes('SCHOOL_ADMIN');
+    }
```
Comment (SodhiA1) [inline backend/src/components/institute/institute.js:253]: I think you can re-use this function hasSchoolAdminRole

File: backend/src/components/institute/institute.js [hunk 2]
```diff
@@ -241,6 +241,103 @@ async function getCachedSchoolBySchoolID(req, res) {
...
+
+    if (hasSchoolAdminRoleAuth) {
+      return res.status(HttpStatus.UNAUTHORIZED).json({
+        message: 'You do not have the required access for this function'
+      });
+    }
```
Comment (SodhiA1) [inline backend/src/components/institute/institute.js:259]: I think the condition would be !hasSchoolAdminRoleAuth
This check can also be combined with hasSchoolAdminRole check

File: frontend/src/components/institute/NewSchoolPage.vue [hunk 3]
```diff
@@ -481,15 +482,15 @@ export default {
...
+          this.setFailureAlert(error.response?.data?.message || error.message);
           console.log(error);
         })
         .finally(() => {
           this.processing = false;
         });
     },
     async schoolCategoryChanged(){
-      if(this.newSchool.categoryCode && this.requiredAuthoritySchoolCategories.includes(this.newSchool.categoryCode)){
+      if(this.newSchool.schoolCategoryCode && this.requiredAuthoritySchoolCategories.includes(this.newSchool.schoolCategoryCode)){
```
Comment (SodhiA1) [inline frontend/src/components/institute/NewSchoolPage.vue:493]: Code merge issue?

**Review by SodhiA1 — CHANGES_REQUESTED** [review]

File: frontend/src/components/institute/NewSchoolPage.vue (hunk 3 above)
Comment (SoLetsDev) [reply frontend/src/components/institute/NewSchoolPage.vue:496]: changed to schoolCategoryCode fixed!

**Review by SodhiA1 — APPROVED** [review]

**Pull request merged**